src/app/_services/base.service.ts

```diff
@@ -0,0 +1,43 @@
+import {Injectable} from '@angular/core';
+import {environment} from "@environments/environment";
+import {Observable, of} from "rxjs";
+import {MESSAGE_TYPES} from "@app/_models";
+import {MessageService} from "@app/_services/message.service";
+
+@Injectable({
+    providedIn: 'root'
+})
+export class BaseService {
+    constructor(private messageService: MessageService) {
+    }
+
+    // Usage: this.baseService.addParams(this.baseService.getURL('course','edit','2'), {'param1':'1','param2':'false'});
```

Comment:
You don't need to have the usage here. Also, the usage should not be like that. This is not human-readable. We will probably use it like this.

```
let url = apiService.getUrl('course', ...)
url = apiService.addParams(url, params)
```